daemon/networkdriver/bridge/driver.go

```diff
@@ -242,18 +355,48 @@ func InitDriver(config *Config) error {
 
 	if config.EnableIpForward {
 		// Enable IPv4 forwarding
-		if err := ioutil.WriteFile("/proc/sys/net/ipv4/ip_forward", []byte{'1', '\n'}, 0644); err != nil {
-			logrus.Warnf("WARNING: unable to enable IPv4 forwarding: %s\n", err)
+		if err := setKernelBoolParam(getForwardingKernelParam(ipv4, ""), true); err != nil {
+			logrus.Warnf("Unable to enable IPv4 forwarding: %v", err)
 		}
 
 		if config.FixedCIDRv6 != "" {
 			// Enable IPv6 forwarding
-			if err := ioutil.WriteFile("/proc/sys/net/ipv6/conf/default/forwarding", []byte{'1', '\n'}, 0644); err != nil {
-				logrus.Warnf("WARNING: unable to enable IPv6 default forwarding: %s\n", err)
+			if err := setKernelBoolParam(getForwardingKernelParam(ipv6, ""), true); err != nil {
+				logrus.Warnf("Unable to enable IPv6 default forwarding: %v", err)
+			}
+			if err := setKernelBoolParam(getForwardingKernelParam(ipv6, "all"), true); err != nil {
+				logrus.Warnf("Unable to enable IPv6 all forwarding: %v", err)
+			}
+		}
+	}
+
+	if !config.InterContainerCommunication && config.EnableIptables {
+		ipVersion := ipv4
+		tryLoad := false
+		if config.FixedCIDR != "" || config.EnableIPv6 {
+			ipVersion |= ipv6
+		}
+		err := enableBridgeNetFiltering(ipVersion, bridgeIface)
+		if err != nil {
+			if ptherr, ok := err.(*os.PathError); ok {
+				if errno, ok := ptherr.Err.(syscall.Errno); ok && errno == syscall.ENOENT {
+					tryLoad = true
+				}
 			}
-			if err := ioutil.WriteFile("/proc/sys/net/ipv6/conf/all/forwarding", []byte{'1', '\n'}, 0644); err != nil {
-				logrus.Warnf("WARNING: unable to enable IPv6 all forwarding: %s\n", err)
+		}
+		if tryLoad {
```

Comment:
`br_netfilter` may be compiled-in (kernels before 3.18.0). Modprobing it before checking for existence of `/proc/sys/net/bridge/*` would generate needless warnings on these kernels.

I'm not opposed to modprobing module before trying to enable filtering, but it would still require a check for existence of those params (which is now handled by the first call to `enableBridgeNetFiltering()`). Is it okay to leave modprobing code in [if !config.InteContainerCommunication ...](https://github.com/docker/docker/pull/11405/files#diff-fb93346b0b06f64fcb85ea4a7de12f8dR373) block?